picking_good_slis.md

```diff
@@ -0,0 +1,92 @@
+
+As a starting point on towards realizing this goal, this document aims to
+provide guidance on selecting the right SLIs for your service.
+
+## Selecting SLIs
+
+For SRE teams looking to start monitoring their services, start by identifying
+SLIs to measure. Focus on measuring the basics to start, can refine them over
+time as you gain more experience operating the service.
+
+A good starting point is to attempt to implement an SLI for each of the four
+"Golden Signals".
+
+## The Four Golden Signals of Monitoring
+
+Google defines the four golden signals well in their [SRE Book][1]: "The four
+golden signals of monitoring are latency, traffic, errors, and saturation. If
+you can only measure four metrics of your user-facing system, focus on these
+four."
+
+We recommend implementing at least one SLI for each of these signals.
+
+### Latency
+
+Latency measures how long it takes your service to handle a request. Examples
+of how to measure this might be:
+  * How long does it take for a page of your browser-based application to load
+  * How long does a database query take
+  * How long does it take to provision some user-requested environment or
+    resource
+
+You will likely want to distinguish between the latency of successful and
+failed requests in your measurements and ultimatley create separate SLOs for
+both.
+
+### Traffic
+
+Traffic measures how many concurrent workloads are being handled by a service
+at a point in time. Examples of how to measure this might be:
+  * How many concurrent users have active sessions in your web app
+  * How many messages per second are published to a message queue
+  * How many requests per second is your API handling
+  * How much network bandwidth is your application serving
+
+### Errors
+
+Errors measures the rate of failed requests to your service. Examples of how
+to measure this might be:
+  * The number of requests to your API that have a 5xx response code
+  * The number of errors seen on the client-side by users of your web app
+  * The rate of dropped packets on a network
+
+### Saturation
+
+Saturation measures the utilization of your service relative to that point at
+which it would be considered "full". Effectively implementing SLIs for
+saturation requires knowledge, or at least a hypothesis, of the capacity
+for your service and the limiting factor. Examples of hor to measure this might
+be:
+
+  * CPU utilization percentage for services that are highly CPU-constrained
+  * Memory utilization percentage for services that are highly Memory
+    constrained
+  * Storage capacity usage percentage
+  * Network bandwidth capacity usage percentage
+  * API request count as a percentage of maximum capacity
+
+Work to identify at least one metric from each of these categories to
+implement as SLIs for your service, then focus on identifying SLOs based on
+them. Finally, expect to add to and refine your SLIs over time as you
+[mature](./sre_maturity.md) on your SRE journey.
```

Comment:
Two points on this:

1. It would be better to move this above the descriptions of the four golden signals.
2. It might just be me but I find it much more useful to think about the SLOs first and then think of the SLIs that can back them. The SLOs should be thought of from the perspective of the end user and what the most important aspects of the level of service is to them. For example, in RHOSAK, we want to have a good first experience to the end users on Kafka creation so we defined a `correctness` SLO on how many Kafkas we expect to be created successfully and a creation `latency` SLO as the time it takes from user creation initiation to a ready Kafka instance is very important.

Reply:
I copied this comment into the Google doc for further discussion.